### PR TITLE
Update roles list after removal and moving of some roles

### PR DESCRIPTION
**Pull request by ysims**
For NUbots/NUbots#907
And NUbots/NUbots#866

### CODE REVIEW

**Review by CMurtagh-LGTM — DISMISSED** [review]

File: src/book/02-system/01-foundations/01-overview.mdx [hunk 1]
```diff
@@ -31,45 +31,39 @@ When adding modules to the list in a role, use the syntax `path::to::module`. Fo
...
-
-In a role, you should have only one platform. Possible platforms are `platform::darwin::HardwareIO`, `platform::darwin::HardwareSimulator`, `platform::Webots` and `platform::Gazebo`.
+| Role Name               | Usage                                                                                                                                                                                                  |
+| ----------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| `fake/visualmesh`       | Test the vision system, such as detection of balls and goals, in a simulated environment.                                                                                                              |
+| `fake/keyboardwalk`     | Test parts of the code without a real robot. Demonstrate basic functionality of the robot. Part of the [Getting Started](/guides/main/getting-started) example since it does not require a real robot. |
```
Comment (LachlanCourt) [inline src/book/02-system/01-foundations/01-overview.mdx:37]: Extremely picky but can we swap `fake/keyboardwalk` and `fake/visualmesh`? Everything else is in alphabetical order except these two

**Review by LachlanCourt — CHANGES_REQUESTED** [review]
These rolenames are used throughout the guides to indicate specific ones to test can you make sure these are updated too please 🙏

File: src/book/03-guides/02-tools/02-webots.mdx [hunk 2]
```diff
@@ -272,32 +272,32 @@ Running the RoboCup scenario requires two things: [the Webots simulator](https:/
...
+   docker container run --rm --network=host -e ROBOCUP_ROBOT_ID=1 -e ROBOCUP_TEAM_COLOR=red -e ROBOCUP_SIMULATOR_ADDR=127.0.1.1:10001 robocup-vhsc-nubots:robocup2021 webots/robocup
+   docker container run --rm --network=host -e ROBOCUP_ROBOT_ID=2 -e ROBOCUP_TEAM_COLOR=red -e ROBOCUP_SIMULATOR_ADDR=127.0.1.1:10002 robocup-vhsc-nubots:robocup2021 webots/robocup
+   docker container run --rm --network=host -e ROBOCUP_ROBOT_ID=3 -e ROBOCUP_TEAM_COLOR=red -e ROBOCUP_SIMULATOR_ADDR=127.0.1.1:10003 robocup-vhsc-nubots:robocup2021 webots/robocup
+   docker container run --rm --network=host -e ROBOCUP_ROBOT_ID=1 -e ROBOCUP_TEAM_COLOR=blue -e ROBOCUP_SIMULATOR_ADDR=127.0.1.1:10021 robocup-vhsc-nubots:robocup2021 webots/robocup
+   docker container run --rm --network=host -e ROBOCUP_ROBOT_ID=2 -e ROBOCUP_TEAM_COLOR=blue -e ROBOCUP_SIMULATOR_ADDR=127.0.1.1:10022 robocup-vhsc-nubots:robocup2021 webots/robocup
+   docker container run --rm --network=host -e ROBOCUP_ROBOT_ID=3 -e ROBOCUP_TEAM_COLOR=blue -e ROBOCUP_SIMULATOR_ADDR=127.0.1.1:10023 robocup-vhsc-nubots:robocup2021 webots/robocup
```
Comment (Crafty15) [inline src/book/03-guides/02-tools/02-webots.mdx:300]: I'm just leaving this as a note. I've updated the webots build tool to search for the roles in sub directories and left it as a full file path and file extension `./b webots build  roles/webots/robocup.role`.
 I can see that we might want to change this, but being explicit is good imo.

**Review by LachlanCourt — APPROVED** [review]
LGTM

File: src/book/03-guides/02-tools/02-webots.mdx (hunk 2 above)
Comment (ysims) [reply src/book/03-guides/02-tools/02-webots.mdx]: updated again in NUbots/NUbots#1046

**Review by Tom0Brien — APPROVED** [review]

**Pull request merged**